test/e2e/cloud-provider-azure.go

```diff
@@ -61,6 +63,8 @@ func InstallCalicoAndCloudProviderAzureHelmChart(ctx context.Context, input clus
 	By("Waiting for Ready cloud-node-manager daemonset pods")
 	for _, ds := range []string{"cloud-node-manager", "cloud-node-manager-windows"} {
 		waitInput := GetWaitForDaemonsetAvailableInput(ctx, clusterProxy, ds, kubesystem, specName)
+		windowsNodeCount, _ := strconv.ParseInt(os.Getenv("WINDOWS_WORKER_MACHINE"), 10, 64)
+		waitInput.PodsRunning = int32(windowsNodeCount) + int32(*input.ConfigCluster.WorkerMachineCount) + int32(*input.ConfigCluster.ControlPlaneMachineCount)
```

Comment:
There is probably a better way to do this... I trying to see if this would work but it doesn't seem to be working as I expected.

Reply:
yeah I'm not sure that would work as it would require the nodes to be up, but this step runs right after the first control plane exists, so we might run into timing issues if we have to wait for all the worker nodes. Maybe there's a way we can run this step later in the e2e steps, after the worker nodes come up and we wait for them?

Reply:
ok, that might be what i was missing...

I am going to split the fix and this update out so we can resolve the upstream windows test failures quickly

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2933